app/controllers/application_controller.rb

```diff
@@ -233,7 +233,19 @@ def check_for_select_portal_user_type
     if request.format && request.format.html? && current_visitor && current_visitor.require_portal_user_type && !current_visitor.has_portal_user_type?
       unless session_sensitive_path
         flash.keep
-        redirect_to portal_user_type_selector_path
+
+        #
+        # TODO If there were some way render the underlying content of
+        # the omniauth_origin url beneath the modal signup popup, that would
+        # be nice.
+        #
+        # Also if this could be passed in, rather than stored in the
+        # session, that would. 
+        #
+        # Otherwise, this is not actually used. 
+        #
+        redirect_to portal_user_type_selector_path(omniauth_origin: session["omniauth.origin"])
```

Comment:
I don't see how this omniauth_origin parameter is use by the portal_user_type_selector controller or view.